http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/monitor/LeoPubsubMessageSubscriber.scala

```diff
@@ -408,73 +421,99 @@ class LeoPubsubMessageSubscriber[F[_]: Timer: ContextShift](
 
   private[monitor] def handleCreateDiskMessage(msg: CreateDiskMessage)(
     implicit ev: ApplicativeAsk[F, AppContext]
-  ): F[Unit] = {
-    val createDisk = for {
+  ): F[Unit] =
+    for {
       ctx <- ev.ask
-      operation <- googleDiskService.createDisk(
-        msg.googleProject,
-        msg.zone,
-        Disk
-          .newBuilder()
-          .setName(msg.name.value)
-          .setSizeGb(msg.size.gb.toString)
-          .setZone(msg.zone.value)
-          .setType(msg.diskType.googleString(msg.googleProject, msg.zone))
-          .setPhysicalBlockSizeBytes(msg.blockSize.bytes.toString)
-          .build()
-      )
-      _ <- persistentDiskQuery.updateGoogleId(msg.diskId, GoogleId(operation.getTargetId), ctx.now).transaction[F]
-      task = computePollOperation
-        .pollZoneOperation(
-          msg.googleProject,
-          msg.zone,
-          OperationName(operation.getName),
-          config.persistentDiskMonitorConfig.create.interval,
-          config.persistentDiskMonitorConfig.create.maxAttempts,
-          None
-        )(
-          persistentDiskQuery.updateStatus(msg.diskId, DiskStatus.Ready, ctx.now).transaction[F].void,
-          F.raiseError(
-            new TimeoutException(s"Fail to create disk ${msg.name.value} in a timely manner")
-          ), //Should save disk creation error if we have error column in DB
-          F.unit
-        )
+      task <- createDisk(msg)
       _ <- asyncTasks.enqueue1(
         Task(
           ctx.traceId,
           task,
-          Some(logError(s"${ctx.traceId.asString} | ${msg.googleProject.value}/${msg.diskId.value}", "creating disk")),
+          Some(
+            logError(s"${ctx.traceId.asString} | ${msg.googleProject.value}/${msg.diskId.value}", "creating disk")
+          ),
           ctx.now
         )
       )
     } yield ()
-    createDisk.handleErrorWith {
-      case e =>
-        for {
-          ctx <- ev.ask
-          _ <- logger.error(e)(s"Failed to create disk ${msg.name.value} in Google project ${msg.googleProject.value}")
-          _ <- persistentDiskQuery.updateStatus(msg.diskId, DiskStatus.Failed, ctx.now).transaction[F]
-        } yield ()
-    }
-  }
+
+  //this returns an F[F[Unit]. It kicks off the google operation, and then return an F containing the async polling task
+  private[monitor] def createDisk(msg: CreateDiskMessage)(
+    implicit ev: ApplicativeAsk[F, AppContext]
+  ): F[F[Unit]] =
```

Comment:
I think having a `sync` param is a bit easier to reason about this code. can we just use a `isSync` param